services/datamanager/builtin/sync_test.go

```diff
@@ -696,3 +731,17 @@ func (noop *noopCloudConnectionService) AcquireConnection(ctx context.Context) (
 func (noop *noopCloudConnectionService) Close(ctx context.Context) error {
 	return nil
 }
+
+func waitUntilNoFiles(dir string) {
+	totalWait := time.Second * 3
```

Comment:
Basically my best guess of the longest this can possibly be blocked on CI! This only takes a few 10s of ms to run locally, but things can block when CI gets busy. So this just gives it plenty of time to wait before saying "yeah, that's never happening, fail". The trade off is basicaly between false negatives (e.g. failing in CI due to lack of resources, as opposed to the code not working), and the time time it takes to run in the failure case (longer we wait before admitting defeat). I figure if the test is failing we don't care that much about how long it takes, so was very conservative here

Reply:
Makes sense, thanks for explaining! Agree on the stance of being conservative